wcfsetup/install/files/lib/acp/form/LanguageAddForm.class.php

```diff
@@ -103,149 +73,105 @@ public function readParameters()
     /**
      * @inheritDoc
      */
-    public function readFormParameters()
-    {
-        parent::readFormParameters();
-
-        if (isset($_POST['countryCode'])) {
-            $this->countryCode = StringUtil::trim($_POST['countryCode']);
-        }
-        if (isset($_POST['languageName'])) {
-            $this->languageName = StringUtil::trim($_POST['languageName']);
-        }
-        if (isset($_POST['languageCode'])) {
-            $this->languageCode = StringUtil::trim($_POST['languageCode']);
-        }
-        if (isset($_POST['sourceLanguageID'])) {
-            $this->sourceLanguageID = \intval($_POST['sourceLanguageID']);
-        }
-        if (isset($_POST['locale'])) {
-            $this->locale = $_POST['locale'];
-        }
-    }
-
-    /**
-     * @inheritDoc
-     */
-    public function validate()
+    protected function createForm()
     {
-        parent::validate();
-
-        // language name
-        if (empty($this->languageName)) {
-            throw new UserInputException('languageName');
-        }
-
-        // country code
-        if (empty($this->countryCode)) {
-            throw new UserInputException('countryCode');
-        }
-
-        // language code
-        $this->validateLanguageCode();
+        parent::createForm();
 
-        // source language id
-        $this->validateSource();
-
-        if (!isset($this->locales[$this->locale])) {
-            throw new UserInputException('locale');
-        }
-    }
+        $this->languages = LanguageFactory::getInstance()->getLanguages();
 
-    /**
-     * Validates the language code.
-     */
-    protected function validateLanguageCode()
-    {
-        if (empty($this->languageCode)) {
-            throw new UserInputException('languageCode');
-        }
-        if (LanguageFactory::getInstance()->getLanguageByCode($this->languageCode)) {
-            throw new UserInputException('languageCode', 'notUnique');
-        }
+        $locales = [
+            '0' => WCF::getLanguage()->get('wcf.global.noSelection'),
+        ];
+        $locales = \array_merge($locales, $this->locales);
```

Comment:
Can you check whether the following works:

```
        $locales = [
            '' => WCF::getLanguage()->get('wcf.global.noSelection'),
            ...$this->locales,
        ];
```

I'd generally avoid reassigning variables, this makes types more clear for analysis tools.